src/components/IconButton/index.tsx

```diff
@@ -0,0 +1,161 @@
+import type { HTMLAttributes, MouseEventHandler, ReactElement } from 'react'
+import { Icon } from '@components'
+import type { IconType } from '@components'
+import styled from '@emotion/styled'
+import type { StyledProps } from '@types'
+import type { Theme } from '@emotion/react'
+import { useState } from 'react'
+
+type IconButtonColor =
+  | 'white'
+  | 'black'
+  | 'gray30'
+  | 'primary'
+  | 'primaryWeak'
+  | 'sub'
+  | 'subWeak'
+
+export interface IconButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  onClick?: MouseEventHandler
+  type: IconType
+  size?: 'small' | 'medium' | 'large'
+  color?: IconButtonColor
+  toggleType?: IconType
+  toggleColor?: IconButtonColor
+  hasShadow?: boolean
+  iconButtonStyle?: 'rounded' | 'square' | 'ghost'
+}
```

Comment:
```suggestion
export interface IconButtonProps extends HTMLAttributes<HTMLButtonElement> {
  onClick?: MouseEventHandler
  icon: IconType
  size?: 'small' | 'medium' | 'large'
  color?: IconButtonColor
  hasShadow?: boolean
  theme?: 'rounded' | 'square' | 'ghost'
}
```

- icon button에서의 type은 button의 타입이라는 의미도 가지고 있을 수 있는 것 같아서 icon으로 변경하는게 어떨까요!?
- iconButton 내부이기 때문에 iconButton이라는 prefix가 붙는 iconButtonStyle보다는 theme나 shape? 이 좋을 것 같습니다!
- toggle같은 경우는 따로 ToggleButton을 만들어서 처리하는게 좋을 것같아요!

Reply:
참고해서 반영하도록 하겠습니다!
아이콘 버튼을 사용해서 토글 버튼 컴포넌트를 추가적으로 만드는 것이 좋을까용 ? ㅎㅎ

Reply:
넹 ><!!

Reply:
[요기](https://github.com/price-offer/offer-design-system/pull/35/commits/0679b5977c191cc0670edb15a87665fd7f29f4c2)에 반영 완료 ! ☺️

Reply:
저기여;;;; 제송한데 커밋 아이디 요상하거든녀;;;;??

Reply:
[무야호 ~](https://github.com/price-offer/offer-design-system/pull/35/commits/b1c2abe8af936bb32a78bba03b1906584ac86479l)

Reply:
오늘 새로 소개드리는 새삥 컴포넌트 ["ToggleButton"](https://github.com/price-offer/offer-design-system/pull/35/commits/8e1a2e6090ec25b559ebbd6def414cd69ee2acb5)